src/qt/guiutil.h

```diff
@@ -417,6 +418,13 @@ namespace GUIUtil
             type);
     }
 
+    /**
+     * Shows a QDialog instance asynchronously, and deletes it on close.
+     */
+    void ShowModalDialogAndDeleteOnClose(
+        QDialog* dialog,
+        Qt::WindowModality modality = Qt::ApplicationModal);
```

Comment:
Why this as an argument when it's not used? If you keep this then the function could be `ShowDialogAndDeleteOnClose`?

Reply:
It could be `Qt::WindowModal` as well.

Reply:
> Why this as an argument when it's not used?

For generality.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/336#issuecomment-859333144).